js/main.js

```diff
@@ -1,330 +0,0 @@
-    } else if(checkZip(inputValue) == true){
-        alert('pick a different zip please')
-        
-    } else {
-        getWeather(inputValue);
-    }
-
-
-  }
-
- 
- // async function making an axios call to weather api
-// how do I get that information out and into a object?
-
-let Weather = {}
-
-async function getWeather(zip) {
-    try {
-        const response = await axios.get(`https://api.openweathermap.org/data/2.5/weather?zip=${zip},US&appid=095fe394e44de9a81a6c44190e364b63`);
-        Weather = response;
-        console.log(Weather)
-        zipArray[cardNum] = zip;
-        cardCreator(bodyDivElem)
-        addContentCard(bodyDivContent)
-        addDataDOM(zip)
-        cardNum++;
-        console.log('cardNum', cardNum)
-
-    } catch (error) {
-      console.error(error);
-      alert("Please enter a valid zip code")
-    }
-  }
-
-  //HelperFunc CheckZip
-  function checkZip(input) {
-    for (const i in zipArray){
-        console.log('ZipArray', i, zipArray[i])
-        if (zipArray[i] == input){
-            return true;
-        }
-
-    }
-
-  }
-
-  // converts kelvin into celcius and farenheit
-// returns kelvin, celcius and farenheit
-function tempConvert(K) {
-    const C = K-273.15;
-    const F = ((K-273.15)*1.8)+32;
-    return [Math.trunc(K),Math.trunc(C),Math.trunc(F)]
-}
-
-function addDataDOM(zip) {
-    const toStr = cardNum.toString()
-    document.getElementById('cardDivx'+toStr).setAttribute('style','width: 18rem;')
-    const img = document.getElementById('cardImgx'+toStr)
-    const icon = Weather.data.weather[0].icon
-    const title = document.getElementById('cardTitlex'+toStr)
-    const text = document.getElementById('cardTextx'+toStr)
-    const tempk = document.getElementById('cardli1x'+toStr)
-    const tempc = document.getElementById('cardli2x'+toStr)
-    const tempf = document.getElementById('cardli3x'+toStr)
-    const temps = tempConvert(Weather.data.main.temp)
-    
-
-    title.innerHTML = Weather.data.name;
-    text.innerHTML = Weather.data.weather[0].main + ', ' + Weather.data.weather[0].description;
-    tempk.innerHTML = tempk.innerHTML + "  " + temps[0]
-    tempc.innerHTML = tempc.innerHTML + "  " + temps[1]
-    tempf.innerHTML = tempf.innerHTML + "  " + temps[2]
-
-    img.setAttribute('src',`http://openweathermap.org/img/wn/${icon}@2x.png`);
-    img.setAttribute('alt',Weather.data.weather[0].main)
-
-    const delBtn = document.getElementById('cardBtnDelx'+toStr);
-    const refBtn = document.getElementById('cardBtnRefx'+toStr);
-    delRefBtn(delBtn,refBtn,zip,toStr)
-
-
-
-    // Image display
-//`http://openweathermap.org/img/wn/${icon}@2x.png`
-}
-
-// click event listeners Update function and delete function
-
-    //delBtn.addEventListener('click',)
-    //refBtn.addEventListener('click',)
-function delRefBtn(del,ref,zip,toStr){
-    del.addEventListener('click',delCard)
-    ref.addEventListener('click',function(){ refCard(zip, toStr); })
```

Comment:
good to see you using a reference to a function AND then needing to pass parameters. Cool. 

this could be:

```javascript
ref.addEventListener('click',() => refCard(zip, toStr))